File: app/app/src/main/java/ca/jame/ontechroom/types/Room.java
package ca.jame.ontechroom.types;

import java.io.Serializable;
import java.util.ArrayList;

public class Room implements Serializable {
    public String name;
    public String imageURL;
    public ArrayList<String> facilities;
    public int floor;
    public int capacity;
    public int minRequired;
    public int maxDuration;
    public String _id;
}

File: server/src/api/Types/Room.ts
import {JSONSchema} from "class-validator-jsonschema";
import {IsArray, IsNumber, IsString} from "class-validator";

@JSONSchema({
    description: 'A Room Object',
    example: {
        id: '',
        name: '',
        imageURL: '',
        facilities: [
            ''
        ],
        floor: 0,
        capacity: 0,
        minRequired: 0,
        maxDuration: 0
    }
})
export class Room {
    @IsString()
    @JSONSchema({
        description: 'The Room ID',
        format: 'LIB####'
    })
    public id: string;
    @IsString()
    public name: string;
    @IsString()
    public imageURL: string;
    @IsArray()
    public facilities: string[];
    @IsNumber()
    public floor: number;
    @IsNumber()
    public capacity: number;
    @IsNumber()
    public minRequired: number;
    @IsNumber()
    public maxDuration: number;
}
